cgroups.rst

```diff
@@ -1,151 +1,237 @@
- Limiting Container Resources with Cgroups
-###########################################
-
-The cgroups (control groups) functionality of the Linux kernel allows
-you to limit and meter the resources used by a process, or group of
-processes. Using cgroups you can limit memory and CPU usage. You can
-also rate limit block IO, network IO, and control access to device
-nodes.
-
-There are two versions of cgroups in common use. Cgroups v1 sets
-resource limits for a process within separate hierarchies per resource
-class. Cgroups v2, the default in newer Linux distributions, implements
-a unified hierarchy, simplifying the structure of resource limits on
-processes.
-
--  v1 documentation:
-   https://www.kernel.org/doc/Documentation/cgroup-v1/cgroups.txt
--  v2 documentation:
-   https://www.kernel.org/doc/Documentation/cgroup-v2.txt
-
-***************************************
- Running {Project} Inside a Cgroup
-***************************************
+##############################
+ Limiting Container Resources
+##############################
 
-Because {Project} starts a container as a simple process, rather
-than using a daemon, you can limit resource usage by running the
-``{command}`` command inside an existing cgroup. This is convenient
-where, for example, a job scheduler uses cgroups to control job limits.
-By running ``{command}`` inside your batch script, your container will
-respect the limits set by the scheduler on the job's cgroup.
+It's often useful to limit the resources that are consumed by a container, e.g.
+to allow the container to only use 1 CPU, or 50% of the RAM on the system.
+Although on HPC clusters it's common to launch containers with a job scheduler
+that can set limits per job, the following scenarios are examples where more
+direct control is useful:
 
-systemd-run
-===========
+* When running multiple containerized applications inside an HPC job, where each
+  container in the job should have different resource limits.
 
-As well as schedulers you can use tools such as ``systemd-run`` to
-create a cgroup, and run {Project} inside of it. This is convenient
-on modern cgroups v2 systems, where the creation of cgroups can be
-delegated to users through systemd. Without this delegation ``root``
-privileges are required to create a cgroup.
+* When testing HPC code on a workstation, to avoid excessive CPU / RAM usage
+  bringing the desktop environment and other applications to a halt.
 
-For example, assuming your system is configured correctly for
-unprivileged cgroup creation via systemd, you can limit the number of
-CPUs a container run is allowed to use:
+* When benchmarking code that doesn't provide internal means to limit the number
+  of CPUs it uses.
 
-.. code::
+There are three ways to apply limits to a container that is run with
+{Project}:
 
-   $ systemd-run --user --scope -p AllowedCPUs=1,2 -- {command} run mycontainer.sif
+* Using the command line flags introduced in v3.10.
```

Comment:
Don't refer to a specific version.  Instead say "Using per-resource command line limit flags described below".